plutus-contract/test/Spec/Emulator.hs

```diff
@@ -139,59 +145,67 @@ selectCoinProp = property $ do
 
 txnUpdateUtxo :: Property
 txnUpdateUtxo = property $ do
-    (Mockchain m _ _, txn) <- forAll genChainTxn
+    (Mockchain m utxo params, txn) <- forAll genChainTxn
     let options = defaultCheckOptions & emulatorConfig . Trace.initialChainState .~ Right m
+        cUtxoIndex = either (error . show) id $ Validation.fromPlutusIndex params $ Index.UtxoIndex utxo
```

Comment:
Shouldn't these 2 lines be moved to `genChainTxn`?

Reply:
Yeah, would be nice, but we tweak the emulator tx a bit after generation for the tests purposes, and there is no ways to tweak the cardano tx in the tests yet. So for now I decided to just duplicate the code. But in the next PRs we can generalize it.